src/battle_ai_main.c

```diff
@@ -2638,6 +2638,23 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
              && AI_WhoStrikesFirst(battlerAtk, battlerDef, move) == AI_IS_SLOWER)
                 score -= 10;
             break;
+        case EFFECT_JUNGLE_HEALING:
+            if (AtMaxHp(battlerAtk)
+             && AtMaxHp(BATTLE_PARTNER(battlerAtk))
+             && !(gBattleMons[battlerAtk].status1 & STATUS1_ANY)
+             && !(gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY))
+                score -= 10;
+            else if (AI_DATA->hpPercents[battlerAtk] >= 90 || AI_DATA->hpPercents[BATTLE_PARTNER(battlerAtk)] >= 90)
```

Comment:
...And what if either the attacker or partner has a status condition? It would be better if that was checked for as well. Take for instance both the attacker and partner are at full HP but are paralyzed. The AI would choose not to use Jungle Healing because of the current logic.

Reply:
That's what
```c
             && !(gBattleMons[battlerAtk].status1 & STATUS1_ANY)
             && !(gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY))
```
is for. if they're both at max health *AND* both it and its partner have no status condition, reduce 10 from the score.

Reply:
Here's how it reads currently. If both the attacker and partner are all full health and neither the attacker nor the partner have a status condition, do `score - 10`. So far so good. But I'm commenting on the next block that is entered if either mon isn't at full health or either mon has a status condition. This second block is saying that if either mon has more than 90% of their HP remaining, do `score - 10`.

So take for instance another case. Attacker is at 1 HP remaining while the partner is at full health. The `AI_DATA->hpPercents[BATTLE_PARTNER(battlerAtk)] >= 90` condition will evaluate to `TRUE` causing `score - 9`. This is clearly not what's wanted.